### PR TITLE
dotCMS/core#24604 fix load-pages-portlet-with-no-active-index

### LINKED ISSUE #24604: When loading a page in edit mode, make the pages favorites star load async.  Right now, if the query fails, the whole page load fails.  You can see this by deactivating you indexes and browsing to a page and try to edit it.
### Parent Issue

https://github.com/dotCMS/core/issues/24326

### Task

When loading a page in edit mode, make the pages favorites star load async. Right now, if the query fails, the whole page load fails. You can see this by deactivating you indexes and browsing to a page and try to edit it

- Current: 
image

- Expected
image

- Current: not loading page 
image

- Expected:
image

### Steps
- Go to `Maintenance` 
- Deactivate Indexes 
image
- Go to `Pages` portlet

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
* When loading a page in edit mode, make the pages favorites star load async. Right now, if the query fails, the whole page load fails. You can see this by deactivating you indexes and browsing to a page and try to edit it

### Screenshots
- Current:  not loading portlet 
image

- Expected
image

- Current: not loading page 
image

- Expected:
image

### CODE REVIEW

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts [hunk 1]
```diff
@@ -219,17 +219,17 @@ export class DotPageStateService {
                         })
                         .pipe(
                             take(1),
+                            catchError((error: HttpErrorResponse) => {
+                                // Deleting message to throw a generic error message
+                                error.error.message = '';
+                                this.dotHttpErrorManagerService.handle(error);
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts:225]: @fmontes by showing the dialog the user knows something weird happened. The actual error from BE is this
image

so unless you think we don't have to show it I can change it?

**Comment by alfredo-dotcms** [comment]
@fmontes changing the alert dialog error with toast
image

File: core-web/apps/dotcms-ui/src/app/api/services/dot-http-error-manager/dot-http-error-manager.service.ts [hunk 2]
```diff
@@ -47,7 +51,9 @@ export class DotHttpErrorManagerService {
      * @returns Observable<boolean>
      * @memberof DotHttpErrorManagerService
      */
-    handle(err: HttpErrorResponse): Observable<DotHttpErrorHandled> {
+    handle(err: HttpErrorResponse, unobtrusive?: boolean): Observable<DotHttpErrorHandled> {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-http-error-manager/dot-http-error-manager.service.ts:54]:
```suggestion
    handle(err: HttpErrorResponse, unobtrusive = false): Observable<DotHttpErrorHandled> {
```

File: core-web/apps/dotcms-ui/src/app/api/services/dot-http-error-manager/dot-http-error-manager.service.ts [hunk 3]
```diff
@@ -87,61 +93,103 @@ export class DotHttpErrorManagerService {
...
-            header: this.dotMessageService.get('dot.common.http.error.403.header')
-        });
+        const header = this.dotMessageService.get('dot.common.http.error.403.header');
+        const message = this.dotMessageService.get('dot.common.http.error.403.message');
+
+        if (this._unobtrusive) {
+            this.showToastMessage(message);
+        } else {
+            this.showAlertDialog(header, message);
+        }
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-http-error-manager/dot-http-error-manager.service.ts:103]: We should move to method, is repeated 6 times.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts [hunk 4]
```diff
@@ -94,12 +94,15 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
                 label: this.dotMessageService.get('All'),
                 value: null
             });
-            languages.forEach((language) => {
-                languageOptions.push({
-                    label: `${language.language} (${language.countryCode})`,
-                    value: language.id
+
+            if (languages) {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:98]:
```suggestion
            if (!!languages.length) {
```

File: dotCMS/src/main/webapp/WEB-INF/messages/Language.properties [hunk 5]
```diff
@@ -5082,6 +5082,7 @@ favoritePage.listing.star.icon.tooltip=Edit Favorite Page
 favoritePage.listing.empty.header=You don't have any favorites yet
 favoritePage.listing.empty.content=Add shortcuts to your most used pages by clicking on the<br />star button on the page
 favoritePage.listing.empty.table=No pages found
+favoritePage.error.fetching.data=Error trying to get Favorite pages data
```
Comment (fmontes) [inline dotCMS/src/main/webapp/WEB-INF/messages/Language.properties:5085]:
```suggestion
favoritePage.error.fetching.data=Failed to retrieve Favorite pages data. Try again later.
```

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 4 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:105]: image

Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:105]: so leaving it as `if (languages.length) {`

File: dotCMS/src/main/webapp/WEB-INF/messages/Language.properties (hunk 5 above)
Comment (alfredo-dotcms) [reply dotCMS/src/main/webapp/WEB-INF/messages/Language.properties:5085]: fixed!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts [hunk 6]
```diff
@@ -115,9 +118,11 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
     readonly languageLabels$: Observable<{ [id: string]: string }> = this.select(
         ({ languages }: DotPagesState) => {
             const langLabels = {};
-            languages.forEach((language) => {
-                langLabels[language.id] = `${language.languageCode}-${language.countryCode}`;
-            });
+            if (languages) {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:121]:
```suggestion
            if (languages?.length) {
```

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 6 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts]: fixed!

**Pull request merged**